Python/pywarpx/fields.py

```diff
@@ -161,568 +184,159 @@ def mesh(self, direction):
-        MultiFab arrays since the arrays include the ghost cells.
-
-        Parameters
-        ----------
-        mfi: MFIter
-            The MFIter instance for the current block,
-
-        starts: sequence
-            The minimum indices of the global slice.
-            These can be negative.
-
-        stops: sequence
-            The maximum indices of the global slice.
-            These can be negative.
-
-        icstart: integer
-            The minimum component index of the global slice.
-            These can be negative.
-
-        icstops: integer
-            The maximum component index of the global slice.
-            These can be negative.
-
-        Returns
-        -------
-        block_slices:
-            The slice of the intersection relative to the block
-
-        global_slices:
-            The slice of the intersection relative to the global array where the data from individual block will go
-        """
-        box = mfi.tilebox()
-        if self.include_ghosts:
-            box.grow(self.mf.n_grow_vect)
-
-        ilo = self._get_indices(box.small_end, 0)
-        ihi = self._get_indices(box.big_end, 0)
-
-        # Add 1 to the upper end to be consistent with the slicing notation
-        ihi_p1 = [i + 1 for i in ihi]
-        i1 = np.maximum(starts, ilo)
-        i2 = np.minimum(stops, ihi_p1)
-
-        if np.all(i1 < i2):
-            block_slices = []
-            global_slices = []
-            for i in range(3):
-                block_slices.append(slice(i1[i] - ilo[i], i2[i] - ilo[i]))
-                global_slices.append(slice(i1[i] - starts[i], i2[i] - starts[i]))
-
-            block_slices.append(slice(icstart, icstop))
-            global_slices.append(slice(0, icstop - icstart))
-
-            return tuple(block_slices), tuple(global_slices)
-        else:
-            return None, None
-
-    def __getitem__(self, index):
-        """Returns slice of the MultiFab using global indexing.
-        The shape of the object returned depends on the number of ix, iy and iz specified, which
-        can be from none to all three. Note that the values of ix, iy and iz are
-        relative to the fortran indexing, meaning that 0 is the lower boundary
-        of the whole domain, and in fortran ordering, i.e. [ix,iy,iz].
-        This allows negative indexing, though with ghosts cells included, the first n-ghost negative
-        indices will refer to the lower guard cells.
-
-        Parameters
-        ----------
-        index: integer, or sequence of integers or slices, or Ellipsis
-            Index of the slice to return
-        """
-        # Note that the index can have negative values (which wrap around) and has 1 added to the upper
-        # limit using python style slicing
-        if index == Ellipsis:
-            index = self.dim * [slice(None)]
-        elif isinstance(index, slice):
-            # If only one slice passed in, it was not wrapped in a list
-            index = [index]
```

Comment:
Perhaps this functionality is worth keeping? This could also allow a workaround for the question of needing to do `[(), (), ()]` for 3d to allow `[()]` to work.

Reply:
Yes, this could be done here, but it's a small enough change in pyamrex, that having a temporary solution here is not needed.

Reply:
Thank you!